src/Registry.js

```diff
@@ -0,0 +1,72 @@
+import * as Utility from './Utility.js'
+
+/**
+ * @type {object}
+ */
+export var descriptors = {
+  /**
+   * @type {function}
+   * @param {object} key
+   * @return {boolean}
+   */
+  has: {
+    value: function (key) {
+      return this.uuid in key
+    }
+  },
+  /**
+   * @type {function}
+   * @param {object} key
+   * @return {any}
+   */
+  get: {
+    value: function (key) {
+      return key[this.uuid]
+    }
+  },
+  /**
+   * @type {function}
+   * @param {object} key
+   * @param {object} value
+   * @return {object}
+   */
+  set: {
+    value: function (key, value) {
+      return Utility.define(key, this.uuid, {value: value, configurable: true}), this
+    }
+  }
+}
+
+/**
+ * @constructor
+ */
+export var constructor = typeof WeakMap !== 'function' ? WeakMap : Utility.extend(function Registry () {
```

Comment:
shouldn't this be `===` instead of `!==`?

Reply:
Yes, artifact of manual testing the fallback implementation